fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+                                    lon1, lat1, lon2, lat2, dlon, dlat,      &
+                                    stdlat1, stdlat2, dx, dy, iau_offset,    &
+                                    nbits
+!
+  use module_fcst_grid_comp,  only: fcstSS => SetServices,                   &
+                                    fcstGrid, numLevels, numSoilLayers,      &
+                                    numTracers, num_diag_sfc_emis_flux,      &
+                                    num_diag_down_flux,                      &
+                                    num_diag_type_down_flux,                 &
+                                    num_diag_burn_emis_flux, num_diag_cmass
+
+  use module_wrt_grid_comp,   only: wrtSS => SetServices
+!
+  use module_cplfields,       only: nExportFields,    exportFields,          &
+                                    exportFieldsList, exportFieldTypes,      &
+                                    exportFieldShare,                        &
+                                    nImportFields,    importFields,          &
+                                    importFieldsList, importFieldTypes,      &
+                                    importFieldShare, importFieldsValid,     &
+                                    queryFieldList,   fillExportFields,      &
+                                    exportData
+  use module_cap_cpl,         only: realizeConnectedCplFields,               &
+                                    clock_cplIntval, diagnose_cplFields
+
+
+  implicit none
+  private
+  public SetServices
+!
+!-----------------------------------------------------------------------
+!
+  type(ESMF_Clock),save                       :: clock_fv3
+
+  type(ESMF_GridComp)                         :: fcstComp
+  type(ESMF_State)                            :: fcstState
+  character(len=esmf_maxstr),allocatable      :: fcstItemNameList(:)
+  type(ESMF_StateItem_Flag), allocatable      :: fcstItemTypeList(:)
+  type(ESMF_FieldBundle),    allocatable      :: fcstFB(:)
+  integer, save                               :: FBCount
+
+  type(ESMF_GridComp),    allocatable         :: wrtComp(:)
+  type(ESMF_State),       allocatable         :: wrtState(:)
+  type(ESMF_FieldBundle), allocatable         :: wrtFB(:,:)
+
+  type(ESMF_RouteHandle), allocatable         :: routehandle(:,:)
+  integer, allocatable                        :: fcstPetList(:)
+
+  logical :: profile_memory = .true.
+
+  character(len=160) :: nuopcMsg
+  integer            :: timeslice = 0
+  integer            :: fcstmype
+  integer            :: dbug = 0
+
+!-----------------------------------------------------------------------
+
+  contains
+
+  !-----------------------------------------------------------------------
+  !------------------- Solo fv3gfs code starts here ----------------------
+  !-----------------------------------------------------------------------
+
+  subroutine SetServices(gcomp, rc)
+
```

Comment:
After creating an earth gridcomponent, this is the first entry point to FV3 from JEDI.

Reply:
I think this will also be something that will be required for imported fields as well. If JEDI is going to drive the atmosphere, it will need to be able to change and import 2D and 3D fields to restart the model.